src/lib/Auth.svelte

```diff
@@ -1,35 +1,49 @@
 <script lang="ts">
-    import {onMount} from "svelte";
+  import { onMount } from "svelte";
 
-    export let authResponse: AuthenticationResponse | null;
+  export let authResponse: AuthenticationResponse | null;
+    export let loggedOut: Boolean = false;
+    let refreshIntervalId: number | undefined = undefined;
 
     onMount(async () => {
-        if(await isLoggedIn() || await canRefresh()) {
-            await refreshAccessToken()
-            await setRefreshAccessTokenInterval()
-            authResponse = await settings.authResponse
+        if (await isLoggedIn() || await canRefresh()) {
+            await refreshAccessToken();
+            refreshIntervalId = await setRefreshAccessTokenInterval();
+            authResponse = await settings.authResponse;
         } else {
-            await login()
+            await login();
         }
-    })
+    });
 
-    async function login(): Promise<string | void> {
-        const envVars = await getEnvVariables()
-        if (!envVars) throw new Error("Env variables not set")
+    export async function login(): Promise<string | void> {
+        const envVars = await getEnvVariables();
+        if (!envVars) throw new Error("Env variables not set");
 
-        return invoke('log_in').then((port) => {
-            const webview = new WebviewWindow('Login', {
+        return invoke("log_in").then((port) => {
+            loggedOut = false;
```

Comment:
Burde ikke denne settes til false etter vi har fått token_exchanged eventen? Ser for meg at noen ikke får logget seg inn og lukker login-vinduet, og pr. nå må de da restarte trøkk.

Ser at det samme skjer hvis man lukker login-vinduet første gangen etter åpning av trøkk, så det er kanskje en helhetlig ting vi skulle sett på. Egen sak?